components/List/src/MDCSelfSizingStereoCell.m

```diff
@@ -207,24 +210,45 @@ - (void)mdc_setAdjustsFontForContentSizeCategory:(BOOL)adjusts {
                                                   object:nil];
   }
 
-  [self adjustFontsForContentSizeCategory];
+  [self adjustFontsForDynamicType];
+}
+
+- (void)mdc_setLegacyFontScaling:(BOOL)mdc_legacyFontScaling {
+  if (mdc_legacyFontScaling == _mdc_legacyFontScaling) {
+    return;
+  }
+
+  _mdc_legacyFontScaling = mdc_legacyFontScaling;
+
+  [self adjustFontsForDynamicType];
 }
 
 // Handles UIContentSizeCategoryDidChangeNotifications
 - (void)contentSizeCategoryDidChange:(__unused NSNotification *)notification {
-  [self adjustFontsForContentSizeCategory];
+  [self adjustFontsForDynamicType];
 }
 
-- (void)adjustFontsForContentSizeCategory {
+- (void)adjustFontsForDynamicType {
   UIFont *titleFont = self.titleLabel.font ?: self.defaultTitleLabelFont;
   UIFont *detailFont = self.detailLabel.font ?: self.defaultDetailLabelFont;
-  if (_mdc_adjustsFontForContentSizeCategory) {
-    titleFont =
-        [titleFont mdc_fontSizedForMaterialTextStyle:MDCFontTextStyleTitle
-                                scaledForDynamicType:_mdc_adjustsFontForContentSizeCategory];
-    detailFont =
-        [detailFont mdc_fontSizedForMaterialTextStyle:MDCFontTextStyleCaption
-                                 scaledForDynamicType:_mdc_adjustsFontForContentSizeCategory];
+  if (self.mdc_adjustsFontForContentSizeCategory) {
+    if (titleFont.mdc_scalingCurve && detailFont.mdc_scalingCurve && !self.mdc_legacyFontScaling) {
```

Comment:
This logic looks surprising: I'd expect the logic to look something like:

```
  if (self.mdc_adjustsFontForContentSizeCategory) {
    if (self.mdc_legacyFontScaling) {
      // Old behavior
    } else if (titleFont.mdc_scalingCurve && detailFont.mdc_scalingCurve) {
      // New behavior
    } else {
      // Do nothing.
    }
```

Or put another way: in the new behavior, if `mdc_adjustsFontForContentSizeCategory` is enabled but no scaling curve is provided, should we do nothing? As it stands, this code will always do something (particularly doing legacy behavior even if `self.mdc_legacyFontScaling` is disabled).

Reply:
Hmmm yeah I guess the rationale was legacy is better than nothing. Updated in edcd9b4 ! Maybe we should go back and update the chips one, it looks like this.